src/Sylius/Component/Core/Promotion/Action/UnitDiscountPromotionActionCommand.php

```diff
@@ -26,12 +28,16 @@
 
 abstract class UnitDiscountPromotionActionCommand implements PromotionActionCommandInterface
 {
-    /** @var FactoryInterface */
-    protected $adjustmentFactory;
+    protected FactoryInterface $adjustmentFactory;
 
-    public function __construct(FactoryInterface $adjustmentFactory)
-    {
+    protected MinimumPriceBasedPromotionAmountCalculatorInterface $minimumPriceBasedPromotionAmountCalculator;
+
+    public function __construct(
+        FactoryInterface $adjustmentFactory,
+        MinimumPriceBasedPromotionAmountCalculatorInterface $minimumPriceBasedPromotionAmountCalculator
```

Comment:
Adding an additional and required argument to the constructor of abstract class is BC break, move this calculation logic to the private method in this class, especially that it looks quite simple